src/CoolWifi.cpp

```diff
@@ -628,3 +628,47 @@ bool CoolWifi::addWifi(String ssid, String pass) {
 
   return (true);
 }
+
+
+/**
+ *  CoolWifi::getExternalIP():
+ *  This method is provided to print the
+ *  public IP of a device to a String
+ */
+String CoolWifi::getExternalIP()
+{
+#if DEBUG == 1
+
+  Serial.println("Entering CoolWifi.addWifi() ");
+
+#endif
+
+  WiFiClient client;
+  String IP;
+  if (!client.connect("api.ipify.org", 80)) {
+    Serial.println(F ("Failed to connect with 'api.ipify.org' !"));
+  }
+  else {
+  int timeout = millis() + 800;
+    client.print("GET /?format=json HTTP/1.1\r\nHost: api.ipify.org\r\n\r\n");
```

Comment:
You don't need to use the `?format=json` here. So you can avoid parsing the JSON later

Reply:
I tried and it doesn't worked the "GET /", I just got Timeout... that's also the reason why I count letters below.